crates/oci-tar-builder/src/bin.rs

```diff
@@ -20,13 +22,84 @@ pub fn main() {
         out_dir = env::current_dir().unwrap();
     }
 
+    if !args.module.is_empty() && args.components.is_some() {
+        println!("Mutually exclusive flags: module and components");
+        return;
+    }
+
+    match args.as_artifact {
+        true => {
+            generate_wasm_artifact(args, out_dir).await.unwrap();
+        }
+        _ => {
+            generate_wasi_image(args, out_dir).unwrap();
+        }
+    }
+}
+
+async fn generate_wasm_artifact(args: Args, out_dir: PathBuf) -> Result<(), anyhow::Error> {
+    println!("Generating wasm artifact");
+
+    let mut builder = Builder::<WasmConfig>::default();
+
+    let (conf, path) = match args.components {
+        Some(path) => {
+            let paths = fs::read_dir(&path).unwrap();
+            if paths.count() != 1 {
+                println!(
+                    "Currently only supports a single component file {:?}",
+                    &path
+                );
+            }
+            let (conf, _) = WasmConfig::from_component(&path, None).await.unwrap();
+            (conf, path)
+        }
+        None => {
+            let module_path = args.module.first().unwrap();
+            let (conf, _) = WasmConfig::from_module(module_path, None).await.unwrap();
+            (conf, module_path.to_string())
+        }
+    };
+
+    builder.add_config(
+        conf,
+        args.repo + "/" + &args.name + ":" + &args.tag,
+        spec::MediaType::Other(oci_wasm::WASM_MANIFEST_CONFIG_MEDIA_TYPE.to_string()),
+    );
+
+    let module_path = PathBuf::from(path);
+    builder.add_layer_with_media_type(&module_path, oci_wasm::WASM_LAYER_MEDIA_TYPE.to_string());
+
+    println!("Creating oci tar file {}", out_dir.clone().display());
+    let f = File::create(out_dir.clone()).unwrap();
+    match builder.build(f) {
+        Ok(_) => println!("Successfully created oci tar file {}", out_dir.display()),
+        Err(e) => {
+            print!(
+                "Building oci tar file {} failed: {:?}",
+                out_dir.display(),
+                e
+            );
+            fs::remove_file(out_dir).unwrap_or(print!("Failed to remove temporary file"));
```

Comment:
question: is `out_dir` a temporary file? I checked the argument and it could come from `env:current_dir()`

Reply:
updated error message here